### PR TITLE
Favicon signature updates

### PR DESCRIPTION
**Pull request by adungo-r7**
## Description
This PR includes updates to improve existing favicon fingerprints and also adds a bunch more new signatures worked on by @cbeek-r7 

**Update Summary:**
**290** new fingerprints
**17** existing fingerprints updated

## Motivation and Context
Improving existing favicon fingerprints and adding new signatures.

## How Has This Been Tested?
- `bundle exec rubocop --parallel --format simple`
- `bundle exec bin/recog_standardize xml/*.xml`
- `bundle exec bin/recog_verify --schema-location xml/fingerprints.xsd --no-warnings xml/*.xml`
- `bundle exec rake tests` (All tests pass)

## Types of changes
- New feature (non-breaking change which adds functionality)

## Checklist:
- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by dtwaddle2-r7 — APPROVED** [review]

File: xml/favicons.xml [hunk 1]
```diff
@@ -411,10 +420,10 @@
   </fingerprint>
 
   <fingerprint pattern="^1c4201c7da53d6c7e48251d3a9680449$">
-    <description>Nagios</description>
+    <description>Nagios (3.0.2)</description>
```
Comment (adfoster-r7) [inline xml/favicons.xml:423]: Just wanted to confirm if this change is correct? It looks like this favicon hasn't changed in 17 years:

```
$ curl -s -L https://raw.githubusercontent.com/NagiosEnterprises/nagioscore/master/html/images/favicon.ico | md5sum
1c4201c7da53d6c7e48251d3a9680449  -
```

So I'm not sure if adding the version here is cromulent 👀

Comment (adungo-r7) [reply xml/favicons.xml:423]: Ah yes, that makes sense I added that there after trying to merge some duplicate entries. Thank you!

File: xml/favicons.xml [hunk 2]
```diff
@@ -473,10 +482,10 @@
   </fingerprint>
 
   <fingerprint pattern="^3ef81fad2a3deaeb19f02c9cf67ed8eb$">
-    <description>DocuWiki</description>
+    <description>DocuWiki (0.0.20080505)</description>
```
Comment (adfoster-r7) [inline xml/favicons.xml:485]: Similar to the above; it feels like we can't mark this as version `0.0.20080505` if the application has used the same favicon for 13 years

```
curl -s -L https://raw.githubusercontent.com/dokuwiki/dokuwiki/master/lib/tpl/dokuwiki/images/favicon.ico | md5sum
3ef81fad2a3deaeb19f02c9cf67ed8eb  -
```

Let me know if I've misunderstood anything :+1:

Comment (adungo-r7) [reply xml/favicons.xml:485]: Apologies, I will double check these updated fingerprints and fix them accordingly. Thanks for pointing these out!

**Comment by adungo-r7** [comment]
@adfoster-r7 I was cross-referencing with other favicon databases, and DocuWiki and Nagios seem to be versioned, but I also don't see any harm with not versioning it?

https://owasp.org/www-community/favicons_database#:~:text=dokuwiki%20(0.0.20080505),MD5%3A%203ef81fad2a3deaeb19f02c9cf67ed8eb

https://owasp.org/www-community/favicons_database#:~:text=nagios%20(3.0.2),MD5%3A%201c4201c7da53d6c7e48251d3a9680449

File: xml/favicons.xml [hunk 3]
```diff
@@ -96,20 +97,22 @@
   </fingerprint>
 
   <fingerprint pattern="^4644f2d45601037b8423d45e13194c93$">
-    <description>Tomcat</description>
+    <description>Apache Tomcat (5.5.26), Alfresco Community</description>
```
Comment (adfoster-r7) [inline xml/favicons.xml:100]: Already talking about this in a DM; but looks like some of these fingerprints might've come from nmap originally https://github.com/nmap/nmap/blob/master/nselib/data/favicon-db#L92C1-L92C33

This is likely another scenario we would have to remove the version details from

File: xml/favicons.xml [hunk 4]
```diff
@@ -95,35 +96,86 @@
...
+  </fingerprint>
+
+  <fingerprint pattern="^a4819787db1dabe1a6b669d5d6df3bfd$">
+    <description>Drupal CMS (2.x-4.x)</description>
+    <example>a4819787db1dabe1a6b669d5d6df3bfd</example>
+    <param pos="0" name="service.vendor" value="Drupal"/>
+    <param pos="0" name="service.product" value="CMS"/>
+    <param pos="0" name="service.version" value="2.x-4.x"/>
+    <param pos="0" name="service.certainty" value="0.5"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:drupal:drupal:4.x"/>
```
Comment (adungo-r7) [inline xml/favicons.xml:178]: I wasn't sure how to best handle the CPE range here so I just used the latest version in the range. Open to any suggestions of course to better interpret this.

**Review by adfoster-r7 — APPROVED** [review]
I think this is a good first pass - there might need to be some extra work developing additional patterns for reporting version ranges, versus claiming an absolute version of software has been detected based on a favicon but that can be iterated on :+1:

**Pull request merged**